driver_go18_test.go

```diff
@@ -247,12 +293,33 @@ func TestContextCancelExec(t *testing.T) {
 		dbt.mustExec("CREATE TABLE test (v INTEGER)")
 		ctx, cancel := context.WithCancel(context.Background())
 
-		// Delay execution for just a bit until db.ExecContext has begun.
+		var longProcess *dbProcess
+		longQuery := "INSERT INTO test VALUES (SLEEP(1))"
+
+		// Delay execution for just a bit until db.ExecContext has begun and get long process ID
+		defer time.AfterFunc(100*time.Millisecond, func() {
+			processList, err := getProcesslist(dbname, dbt.db)
+			if err != nil {
+				dbt.Errorf("failed to get mysql process list %v", err)
+			}
+			longProcess = getQueryProcess(processList, longQuery)
+		}).Stop()
+
+		// Delay execution for just a bit until db.ExecContext has begun and cancel context
 		defer time.AfterFunc(100*time.Millisecond, cancel).Stop()
```

Comment:
you can cancel query immediately after get query id.

Reply:
ok